make test fails if you have a local minio server running

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
Tests should proceed regardless of he local environment and shouldn't cause port conflicts on local environment.

<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
Run minio server `docker run --rm -p 9000:9000 minio/minio server /export` and then run `make test`
```
$ make test
Check deps
Checking project is in GOPATH
Installing golint
Installing gocyclo
Installing deadcode
Installing misspell
Installing ineffassign
Running vet
Running fmt
Running lint
Running cyclo
Building minio to /home/harsha/mygo/src/github.com/minio/minio/minio ...
Running unit tests
ok  	github.com/minio/minio	0.014s
ok  	github.com/minio/minio/cmd	86.517s
ok  	github.com/minio/minio/pkg/cgroup	0.003s
ok  	github.com/minio/minio/pkg/disk	0.007s
--- FAIL: TestNewHTTPListener (0.04s)
	listener_test.go:247: error: expected = listen tcp 93.184.216.34:9000: bind: cannot assign requested address, got = listen tcp 127.0.0.1:9000: bind: address already in use
error: junk message.  Received non-HTTP message from new connection 127.0.0.1:50368 at server 127.0.0.1:43598
error: junk message.  Received non-HTTP message from new connection 10.0.0.13:46716 at server 10.0.0.13:40427
error: tls: oversized record received with length 20037.  TLS handshake failed with new connection 127.0.0.1:58060 at server 127.0.0.1:34479
error: tls: oversized record received with length 20037.  TLS handshake failed with new connection 10.0.0.13:50142 at server 10.0.0.13:41150
error: junk message.  Received non-HTTP message from new connection 127.0.0.1:59460 at server 127.0.0.1:46678
error: junk message.  Received non-HTTP message from new connection 10.0.0.13:37864 at server 10.0.0.13:34907
FAIL
```
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Do not use port 9000 while testing `pkg/http` 
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Explained above.

## Context
Consistent testing.
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
master

--- a/pkg/http/listener_test.go
+++ b/pkg/http/listener_test.go
@@ -188,11 +188,11 @@ func TestIsHTTPMethod(t *testing.T) {
 func TestNewHTTPListener(t *testing.T) {
 	errMsg := ": no such host"
 
-	remoteAddrErrMsg := "listen tcp 93.184.216.34:9000: bind: cannot assign requested address"
+	remoteAddrErrMsg := "listen tcp 93.184.216.34:65432: bind: cannot assign requested address"
 	if runtime.GOOS == "windows" {
-		remoteAddrErrMsg = "listen tcp 93.184.216.34:9000: bind: The requested address is not valid in its context."
+		remoteAddrErrMsg = "listen tcp 93.184.216.34:65432: bind: The requested address is not valid in its context."
 	} else if runtime.GOOS == "darwin" {
-		remoteAddrErrMsg = "listen tcp 93.184.216.34:9000: bind: can't assign requested address"
+		remoteAddrErrMsg = "listen tcp 93.184.216.34:65432: bind: can't assign requested address"
 	}
 
 	tlsConfig := getTLSConfig(t)
@@ -208,12 +208,12 @@ func TestNewHTTPListener(t *testing.T) {
 		errorLogFunc           func(error, string, ...interface{})
 		expectedErr            error
 	}{
-		{[]string{"93.184.216.34:9000"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New(remoteAddrErrMsg)},
-		{[]string{"example.org:9000"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New(remoteAddrErrMsg)},
+		{[]string{"93.184.216.34:65432"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New(remoteAddrErrMsg)},
+		{[]string{"example.org:65432"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New(remoteAddrErrMsg)},
 		{[]string{"unknown-host"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New("listen tcp: address unknown-host: missing port in address")},
-		{[]string{"unknown-host:9000"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New("listen tcp: lookup unknown-host" + errMsg)},
-		{[]string{"localhost:9000", "93.184.216.34:9000"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New(remoteAddrErrMsg)},
-		{[]string{"localhost:9000", "unknown-host:9000"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New("listen tcp: lookup unknown-host" + errMsg)},
+		{[]string{"unknown-host:65432"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New("listen tcp: lookup unknown-host" + errMsg)},
+		{[]string{"localhost:65432", "93.184.216.34:65432"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New(remoteAddrErrMsg)},
+		{[]string{"localhost:65432", "unknown-host:65432"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, errors.New("listen tcp: lookup unknown-host" + errMsg)},
 		{[]string{"localhost:0"}, nil, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, nil},
 		{[]string{"localhost:0"}, tlsConfig, time.Duration(0), time.Duration(0), time.Duration(0), nil, nil, nil, nil},
 	}

